prefix_test.go

```diff
@@ -686,3 +1148,62 @@ func TestGob(t *testing.T) {
 		require.Equal(t, prefix, newPrefix)
 	})
 }
+
+func TestPrefix_availablePrefixes(t *testing.T) {
+	tests := []struct {
+		name                   string
+		cidr                   string
+		availableChildPrefixes map[string]bool
+		want                   uint64
+	}{
+		{
+			name:                   "one child prefix",
+			cidr:                   "192.168.0.0/20",
+			availableChildPrefixes: map[string]bool{"192.168.0.0/22": false},
+			want:                   512 + 256,
+		},
+		{
+			name:                   "two child prefixes",
+			cidr:                   "192.168.0.0/16",
+			availableChildPrefixes: map[string]bool{"192.168.0.0/22": false, "192.168.0.0/26": false},
+			want:                   8192 + 4096 + 2048 + 1024 + 512 + 256,
+		},
+		{
+			name:                   "four child prefixes",
+			cidr:                   "192.168.0.0/16",
+			availableChildPrefixes: map[string]bool{"192.168.0.0/22": false, "192.168.0.0/26": false, "192.168.128.0/26": false, "192.168.196.0/26": false},
+			want:                   4096 + 3*2048 + 3*1024 + 3*512 + 3*256 + 2*128 + 2*64 + 2*32 + 2*16,
+		},
+		{
+			name: "simple ipv6",
+			cidr: "2001:0db8:85a3::/120",
+			want: 64,
+		},
+		{
+			name:                   "one child prefix ipv6",
+			cidr:                   "2001:0db8:85a3::/120",
+			availableChildPrefixes: map[string]bool{"2001:0db8:85a3::/122": false},
+			want:                   32 + 16,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			p := &Prefix{
+				Cidr:                   tt.cidr,
+				availableChildPrefixes: tt.availableChildPrefixes,
+			}
+			got, avpfxs := p.availablePrefixes()
+			for _, pfx := range avpfxs {
+				// Only logs if fails
+				ipprefix, err := netaddr.ParseIPPrefix(pfx)
+				require.NoError(t, err)
+				smallest := 1 << (ipprefix.IP.BitLen() - 2 - ipprefix.Bits)
+				t.Logf("available prefix:%s smallest left:%d", pfx, smallest)
+			}
+
+			if diff := deep.Equal(tt.want, got); diff != nil {
```

Comment:
You can simply write `if tt.want != got {` since both variables are uint64

Reply:
will fix

Reply:
done